examples/dns/resource_record_set.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: dns.gcp.crossplane.io/v1alpha1
+kind: ResourceRecordSet
+metadata:
+  name: example
+  annotations:
+    crossplane.io/external-name: "example.crossplane.io."
```

Comment:
I believe with default [NewNameAsExternal](https://github.com/crossplane/crossplane-runtime/blob/ba474e81c62c8e92a810d471a9a04efa62245817/pkg/reconciler/managed/api.go#L46) initializer, we will always have to manually add external name annotation since setting `metadata.name` as `example.crossplane.io.` is not possible:

```
The ResourceRecordSet "example.crossplane.io." is invalid: metadata.name: Invalid value: "example.crossplane.io.": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Do you think if it makes sense to initialize the external name annotation (if not provided) by appending a `.` at the end of `metadata.name` ?

Reply:
Yes, it really does. The correct way to do it is by using a custom initializer in the `Setup` method?

Reply:
Yes, that's correct.